From 7afb6c2fa78a98192e34f0cfe4fd7d25a894799f Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:53 -0400
Subject: [ethernet] mlx4: Using positive error value for unsigned

Message-id: <62e6c0715d3865304cd839bcac032ae9eae25b16.1410425016.git.avadai@redhat.com>
Patchwork-id: 93646
O-Subject: [Patch RHEL7.1 V1 117/197] net/mlx4_en: Using positive error value for unsigned
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Eyal Perry <eyalpe@mellanox.com>

Using a positive value for error: MLX4_NET_TRANS_RULE_NUM instead
of -EPROTONOSUPPORT, to remove compilation warning.

Signed-off-by: Eyal Perry <eyalpe@mellanox.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit c3ca5205e67cfbb0022f004a4036351f90637171)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
index 9090643..15c8811 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_netdev.c
@@ -130,7 +130,7 @@ static enum mlx4_net_trans_rule_id mlx4_ip_proto_to_trans_rule_id(u8 ip_proto)
  case IPPROTO_TCP:
   return MLX4_NET_TRANS_RULE_ID_TCP;
  default:
-  return -EPROTONOSUPPORT;
+  return MLX4_NET_TRANS_RULE_NUM;
  }
 };
 
@@ -177,7 +177,7 @@ static void mlx4_en_filter_work(struct work_struct *work)
  int rc;
  __be64 mac_mask = cpu_to_be64(MLX4_MAC_MASK << 16);
 
- if (spec_tcp_udp.id < 0) {
+ if (spec_tcp_udp.id >= MLX4_NET_TRANS_RULE_NUM) {
   en_warn(priv, "RFS: ignoring unsupported ip protocol (%d)\n",
    filter->ip_proto);
   goto ignore;
-- 
1.7.1